linera-faucet/server/src/lib.rs

```diff
@@ -47,9 +50,11 @@ pub struct QueryRoot<C> {
 
 /// The root GraphQL mutation type.
 pub struct MutationRoot<C> {
-    chain_id: ChainId,
+    main_chain_id: ChainId,
+    tmp_chain_id: Arc<Mutex<Option<ChainId>>>,
```

Comment:
IMHO these two fields should have appropriate comments. `tmp_` and `main_` do not say anything.